### PR TITLE
Database integration

### PR DESCRIPTION
**Pull request by darosior**
This builds on #10 and integrates an sqlite database.

I finally went for sqlite for the laptop wallet, and postgre for the servers which IMO makes much more sense.

The schema is straightforward:
- 4 tables:
    - VERSION: obv
    - WALLETS: stores the Miniscript descriptors, and a timestamp for rescanning
    - VAULTS: stores vaults with their deposit txid, status, etc
    - TRANSACTIONS: stores all transactions as PSBTs, related to a vault
- 2 indexes:
    - vault by status
    - transactions by vault

Still hand-testing.. we'll need a functional testing framework soon..

### CODE REVIEW

**Comment by darosior** [comment]
I wonder if i should put the `our_xpub` for each of them in the database too..

File: src/database/interface.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+) -> Result<Vec<rusqlite::Result<T>>, DatabaseError>
+where
+    P: IntoIterator,
+    P::Item: ToSql,
+    F: FnMut(&Row<'_>) -> rusqlite::Result<T>,
+{
+    let conn = Connection::open(path)
+        .map_err(|e| DatabaseError(format!("Opening database for query: {}", e.to_string())))?;
+
+    let x = Ok(conn
```
Comment (edouardparis) [inline src/database/interface.rs:46]: `let x =` is not necessary I think

**Comment by darosior** [comment]
Rebased on master after #10 merge

**Comment by darosior** [comment]
Added a test for the database creation.

**Comment by edouardparis** [comment]
ACK 235733807cee1ec4cc9b1af96ea45299ee569b0f

**Review by edouardparis — COMMENTED** [review]
ACK 235733807cee1ec4cc9b1af96ea45299ee569b0f

**Pull request merged**